video/audio_sample.h

```diff
@@ -9,35 +9,36 @@
 #include "buffer_stream.h"
 
 struct AudioSample {
-	AudioSample(std::vector<std::shared_ptr<BufferStream>>& streams, uint8_t format, uint32_t sampleRate = AUDIO_DEFAULT_SAMPLE_RATE, uint16_t frequency = 0) :
-		blocks(streams), format(format), sampleRate(sampleRate), baseFrequency(frequency) {}
+	AudioSample(std::vector<std::shared_ptr<BufferStream>>& streams, uint16_t id, uint8_t format, uint32_t sampleRate = AUDIO_DEFAULT_SAMPLE_RATE, uint16_t frequency = 0) :
+		blocks(streams), id(id), format(format), sampleRate(sampleRate), baseFrequency(frequency) {}
 	~AudioSample();
 
 	int8_t getSample(uint32_t & index, uint32_t & blockIndex);
 	void seekTo(uint32_t position, uint32_t & index, uint32_t & blockIndex, int32_t & repeatCount);
 	uint32_t getSize();
 
 	std::vector<std::shared_ptr<BufferStream>>& blocks;
+	uint16_t		id;					// ID of the sample
 	uint8_t			format;				// Format of the sample data
 	uint32_t		sampleRate;			// Sample rate of the sample
 	uint16_t		baseFrequency = 0;	// Base frequency of the sample
 	int32_t			repeatStart = 0;	// Start offset for repeat, in samples
 	int32_t			repeatLength = -1;	// Length of the repeat section in samples, -1 means to end of sample
-	std::unordered_map<uint8_t, std::weak_ptr<AudioChannel>> channels;	// Channels playing this sample
+	// std::unordered_map<uint8_t, std::weak_ptr<AudioChannel>> channels;	// Channels playing this sample
 };
 
 AudioSample::~AudioSample() {
 	// iterate over channels
-	for (auto channelPair : this->channels) {
```

Comment:
as noted above, channel tracking on samples has been removed

Reply:
the main effect of removing channel tracking on samples is that when a sample is "cleared" (or its buffer deleted) previously that would have ensured that any channel using that sample would get set back to the default sound

now what happens is the channel keeps working and the sample object will be retained - it's data is just no longer associated with a buffer, and it cannot be selected for use on another channel.  once the channel is changed to use a different waveform, the sample data will be discarded